packages/core/src/types/IKeyManager.ts

```diff
@@ -81,6 +81,35 @@ export interface IKeyManagerDecryptJWEArgs {
   data: string
 }
 
+/**
+ * Input arguments for {@link IKeyManager.keyManagerSign | keyManagerSign}
+ * @public
+ */
+export interface IKeyManagerSignArgs {
+  /**
+   * Key ID
+   */
+  kid: string
+
+  /**
+   * The algorithm to use for signing.
+   * This must be one of the algorithms supported by the KMS for this key type.
+   */
+  alg?: string
+
+  /**
+   * Data to sign
+   */
+  data: string
```

Comment:
Should this be rather a byte array? Then, the encoding parameter would be more powerful.

Reply:
If this were a byte array then the encoding param would not be needed. The encoding param is used to reason about the input data.
The constraint is that plugin methods need to be exposed over OpenAPI as well, and there is no support for byte arrays there.
I chose the string plus encoding approach to satisfy both open API and various input formats.

It's the responsibility of the `key-manager` plugin to convert the data string into a `UInt8Array` based on the encoding specified and to forward this to the KMS implementations, which do not have this OpenAPI constraint any more.